sparkmagic/sparkmagic/magics/sparkmagicsbase.py

```diff
@@ -40,6 +41,12 @@ def __init__(self, shell, data=None, spark_events=None):
     def execute_spark(self, cell, output_var, samplemethod, maxrows, samplefraction, session_name, coerce):
         (success, out) = self.spark_controller.run_command(Command(cell), session_name)
         if not success:
+            if conf.spark_statement_errors_are_fatal():
+                if conf.shutdown_session_on_spark_statement_errors():
```

Comment:
This seems like a good check. Do be you know if the session is cleaned up when the process exits in ideal circumstances? If so this might be unnecessary as the process / kernel exit would resolve Livy cleanup.

Reply:
It does not, the Livy session stays open and available.